crates/ruff_python_parser/src/lexer.rs

```diff
@@ -780,6 +780,17 @@ impl<'source> Lexer<'source> {
 
                 self.lex_magic_command(kind)
             }
+
+            '?' if self.mode == Mode::Jupyter
+                && !self.state.is_new_logical_line()
+                && matches!(
+                    [self.cursor.first(), self.cursor.second()],
+                    ['?', '\n' | '\r' | EOF_CHAR] | ['\n' | '\r' | EOF_CHAR, _]
+                ) =>
```

Comment:
I'm not sure if we should be this strict in the lexer. This just makes sure that the `Question` token is emitted only when it's at the end of the line:

```python
foo?
foo??
```

Reply:
I think we can always emit it in Jupyter mode. It's a valid token and it's the Parser's job to determine if it is a valid position.